open_spiel/games/maedn.cc

```diff
@@ -0,0 +1,589 @@
+// Copyright 2022 Thorsten Jungblut <ulli@netcorner.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "open_spiel/games/maedn.h"
+
+#include <algorithm>
+#include <cstdlib>
+#include <set>
+#include <utility>
+#include <vector>
+
+#include "open_spiel/abseil-cpp/absl/strings/str_cat.h"
+#include "open_spiel/game_parameters.h"
+#include "open_spiel/spiel.h"
+#include "open_spiel/spiel_utils.h"
+
+namespace open_spiel {
+namespace maedn {
+namespace {
+
+const std::vector<std::pair<Action, double>> kChanceOutcomes = {
+    std::pair<Action, double>(0, 1.0 / 6),
+    std::pair<Action, double>(1, 1.0 / 6),
+    std::pair<Action, double>(2, 1.0 / 6),
+    std::pair<Action, double>(3, 1.0 / 6),
+    std::pair<Action, double>(4, 1.0 / 6),
+    std::pair<Action, double>(5, 1.0 / 6),
+};
+
+const std::vector<int> kChanceOutcomeValues =
+    {1, 2, 3, 4, 5, 6};
+
+int NumPiecesPerPlayer(const Game* game) {
+  return kNumPiecesPerPlayer;
+}
+
+// Facts about the game
+const GameType kGameType{
+    /*short_name=*/"maedn",
+    /*long_name=*/"Mensch-Aergere-Dich-Nicht",
+    GameType::Dynamics::kSequential,
+    GameType::ChanceMode::kExplicitStochastic,
+    GameType::Information::kPerfectInformation,
+    GameType::Utility::kZeroSum,
+    GameType::RewardModel::kTerminal,
+    /*max_num_players=*/4,
+    /*min_num_players=*/2,
+    /*provides_information_state_string=*/false,
+    /*provides_information_state_tensor=*/false,
+    /*provides_observation_string=*/true,
+    /*provides_observation_tensor=*/true,
+    /*parameter_specification=*/
+    {
+      // Number of Players (2 to 4)
+      {"players", GameParameter(2)},
+      // If two players play, two different settings are possible:
```

Comment:
Please move this description to header file. See comment there regarding params.